contracts/crowdsale/ZxcCrowdsale.sol

```diff
@@ -0,0 +1,278 @@
+pragma solidity ^0.4.24;
+
+import "@0xcert/ethereum-utils/contracts/math/SafeMath.sol";
+import "@0xcert/ethereum-utils/contracts/ownership/Ownable.sol";
+import "@0xcert/ethereum-zxc/contracts/tokens/Zxc.sol";
+
+
+/**
+ * @title ZXC crowdsale contract.
+ * @dev Crowdsale contract for distributing ZXC tokens.
+ */
+contract ZxcCrowdsale is Ownable {
+  using SafeMath for uint256;
+
+  /**
+   * Token being sold.
+   */
+  Zxc public token;
+
+  /**
+   * Start timestamps for the following stages (start dates are inclusive, end exclusive):
+   * - Token pre-sale with 10% bonus: 2018/06/26 - 2018/07/04
+   * - Token sale with 5% bonus: 2018/07/04 - 2018/07/05
+   * - Token sale with 0% bonus: 2018/07/05 - 2018/07/18
+   */
+  uint256 public startTimePresale;
+  uint256 public startTimeSaleWithBonus;
+  uint256 public startTimeSaleNoBonus;
+
+  uint256 public bonusPresale;
```

Comment:
Yes. I think there's another issue to discuss: if bonus is e.g. 33% then: 100%/33% = 3.0303030303. This will cut the decimals, so we might want to perform this calculation directly where we calculate the bonus token amount: instead of `tokens.div(bonusPresale)` we do `tokens.mul(uint256(100).div(bonusPresalePercentage)`. Wdyt?

Reply:
I agree.

Reply:
5acbecc